rsocket-core/src/main/java/io/rsocket/RSocketRequester.java

```diff
@@ -183,6 +187,11 @@ public boolean isDisposed() {
       return Mono.error(err);
     }
 
+    if (!FragmentationUtils.isValid(this.mtu, payload)) {
+      payload.release();
+      return Mono.error(new IllegalArgumentException("Too big Payload size"));
```

Comment:
I would tend to reverse the sentence in the error message: `Payload size too big`. Maybe also allude to where the limitation comes from (not necessarily giving the value of the `mtu`, more like how it is configured on the factory, ie `Payload size too big compared to the configured fragment size`, or something like that, since `mtu` is set by the `fragment(int)` method).

(this comment applies to all the instances of the message below as well)

Reply:
Agreed, perhaps something like:
> The payload is too big to send as a single frame with a 24-bit encoded length. Considering enabling fragmentation via RSocketFactory.

Reply:
fixed